perses/tests/test_topology_proposal.py

```diff
@@ -0,0 +1,440 @@
+import simtk.openmm.app as app
+from simtk.openmm.app.internal.pdbstructure import PdbStructure
+import copy
+import numpy as np
+try:
+    from urllib.request import urlopen
+    from io import StringIO
+except:
+    from urllib2 import urlopen
+    from cStringIO import StringIO
+
+
+def load_pdbid_to_openmm(pdbid):
+    """
+    create openmm topology without pdb file
+    lifted from pandegroup/pdbfixer
+    """
+    url = 'http://www.rcsb.org/pdb/files/%s.pdb' % pdbid
+    file = urlopen(url)
+    contents = file.read().decode('utf-8')
+    file.close()
+    file = StringIO(contents)
+
+    if _guessFileFormat(file, url) == 'pdbx':
+        pdbx = app.PDBxFile(contents)
+        topology = pdbx.topology
+        positions = pdbx.positions
+    else:    
+        structure = PdbStructure(file)
+        pdb = app.PDBFile(structure)
+        topology = pdb.topology
+        positions = pdb.positions
+
+    return topology, positions
+
+def _guessFileFormat(file, filename):
+    """
+    Guess whether a file is PDB or PDBx/mmCIF based on its filename and contents.
+    authored by pandegroup
+    """
+    filename = filename.lower()
+    if '.pdbx' in filename or '.cif' in filename:
+        return 'pdbx'
+    if '.pdb' in filename:
+        return 'pdb'
+    for line in file:
+        if line.startswith('data_') or line.startswith('loop_'):
+            file.seek(0)
+            return 'pdbx'
+        if line.startswith('HEADER') or line.startswith('REMARK') or line.startswith('TITLE '):
+            file.seek(0)
+            return 'pdb'
+    # It's certainly not a valid PDBx/mmCIF.  Guess that it's a PDB.
+    file.seek(0)
+    return 'pdb'
+
+def _matchResidue(res, template, bondedToAtom):
```

Comment:
The failure to find a match is happening inside here somewhere, which is why I copied it (originally was using app.forcefield._matchResdiue, as seen commented out in Line 154)